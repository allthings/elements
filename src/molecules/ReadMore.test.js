import React from 'react'
import { Text, ReadMore } from '../'

class ReadMoreTest extends ReadMore {
    childRef = { current: { style: { height: 480 } } }
}

describe('ReadMore component', () => {
  test('ReadMoreTest renders properly width a px height of 20', () => {
    const wrapper = mount(
      <ReadMoreTest defaultHeight={20}>
        <Text>Testing a short text...</Text>
      </ReadMoreTest>
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('ReadMoreTest renders properly width a vh height of 10vh', () => {
    const wrapper = mount(
      <ReadMoreTest defaultHeight="10vh">
        <Text>Testing a short text with 10vh...</Text>
      </ReadMoreTest>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
