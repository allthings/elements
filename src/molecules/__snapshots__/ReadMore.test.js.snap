// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMore component ReadMoreTest renders properly width a px height of 20 1`] = `
.css-0,
[data-css-0] {
  width: 100%;
}

.css-1,
[data-css-1] {
  transition-property: height;
  transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  overflow: hidden;
  transform-origin: top;
  -webkit-transition-property: height;
  -moz-transition-property: height;
  -webkit-transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -moz-transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -webkit-transform-origin: top;
}

.css-2,
[data-css-2] {
  box-sizing: border-box;
  align-content: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  flex: 0 0 auto;
  -webkit-align-content: stretch;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -webkit-flex: 0 0 auto;
}

.css-3,
[data-css-3] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-4,
[data-css-4] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #333333;
}

<div
  data-css-0=""
  data-css-2=""
>
  <div
    data-css-3=""
    data-css-1=""
  >
    <div
      data-css-4=""
      data-css-3=""
    >
      Testing a short text...
    </div>
  </div>
</div>
`;

exports[`ReadMore component ReadMoreTest renders properly width a vh height of 10vh 1`] = `
.css-0,
[data-css-0] {
  width: 100%;
}

.css-1,
[data-css-1] {
  transition-property: height;
  transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  overflow: hidden;
  transform-origin: top;
  -webkit-transition-property: height;
  -moz-transition-property: height;
  -webkit-transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -moz-transition: height 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  -webkit-transform-origin: top;
}

.css-2,
[data-css-2] {
  box-sizing: border-box;
  align-content: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  flex: 0 0 auto;
  -webkit-align-content: stretch;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -webkit-flex: 0 0 auto;
}

.css-3,
[data-css-3] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-4,
[data-css-4] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-size: 14px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #333333;
}

<div
  data-css-0=""
  data-css-2=""
>
  <div
    data-css-3=""
    data-css-1=""
  >
    <div
      data-css-4=""
      data-css-3=""
    >
      Testing a short text with 10vh...
    </div>
  </div>
</div>
`;
